channel/mock_app.go

```diff
@@ -61,14 +62,21 @@ func NewMockOp(op MockOp) *MockOp {
 	return &op
 }
 
-// Encode encodes a MockOp into an io.Writer.
-func (o MockOp) Encode(w io.Writer) error {
-	return perunio.Encode(w, uint64(o))
+// MarshalBinary marshals MockOp to its binary representation.
+func (o MockOp) MarshalBinary() ([]byte, error) {
+	data := make([]byte, mockOpBinaryLength)
+	binary.PutUvarint(data, uint64(o))
```

Comment:
Use `binary.BigEndian.PutUint64` instead?

Reply:
Yes, I think constant length encoding is better, as it is simpler to replicate across implementations. Updated in 0dc3cca7.